dpnp/backend/kernels/dpnp_krnl_logic.cpp

```diff
@@ -52,6 +52,28 @@ void dpnp_all_c(void* array1_in, void* result1, const size_t size)
     return;
 }
 
+template <typename _DataType, typename _ResultType>
```

Comment:
again, do not understand why do you templating output type? You have instances only for `bool`.

Reply:
I did a little research myself.
And I found out that there is an `out` param which can be different from `np.boolean`:
See:
```python
>>> o=np.array(False).astype(np.uint8)
>>> z=np.any([-1, 4, 5], out=o)
>>> z
array(1, dtype=uint8)
>>> id(z)
140070830227248
>>> id(o)
140070830227248
```
Therefore, we can save the templating output for future implementation and enabling `out` param.